nix/download.tt

```diff
@@ -4,23 +4,30 @@
 
 <p>The latest and recommended version of Nix is
 <strong>[%latestNixVersion%]</strong>. The quickest way to install it
-on Linux and Mac OS X is to run the following in a shell (as a user
+on Linux and macOS is to run the following in a shell (as a user
 other than <tt>root</tt>):
 
-<pre><span class="muted">$</span> curl <a href="[%root%]nix/install">https://nixos.org/nix/install</a> | sh</pre>
+<pre>
+<span class="command">curl -o install-nix-[%latestNixVersion%] https://nixos.org/nix/install</span>
+<span class="command">echo "[%nixInstallHashsha256%]  install-nix-[%latestNixVersion%]" | (shasum -p 256 || sha256sum -c) 2> /dev/null</span>
+<span class="command">sh install-nix-[%latestNixVersion%]</span>
+</pre>
```

Comment:
Tragically, macOS only has shasum out of the box, where as most linuxes have sha256sum, and some don't have shasum. For super bonus points, some don't have either and you have to use openssl >.> I'm omitting support for thaht, because I'm going to assume they can figure out how to check a hash (or not.)